sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1798,15 +1821,31 @@ ProgramManager::kernelImplicitLocalArgPos(const std::string &KernelName) const {
   return {};
 }
 
+static bool skipEmptyImage(sycl_device_binary RawImg) {
+  // bfloat16 devicelib image must be kept.
+  sycl_device_binary_property_set ImgPS;
+  for (ImgPS = RawImg->PropertySetsBegin; ImgPS != RawImg->PropertySetsEnd;
+       ++ImgPS) {
+    if (ImgPS->Name &&
+        !strcmp(__SYCL_PROPERTY_SET_DEVICELIB_BF16_TYPE, ImgPS->Name)) {
+      return false;
+    }
+  }
+
+  return true;
```

Comment:
Nit, not that it's that much smaller:
```suggestion
return !std::any_of(RawImg->PropertySetsBegin, RawImg->PropertySetsEnd,
                    [](const _sycl_device_binary_property_set_struct &ImgPS) {
                      return ImgPS->Name &&
                             !strcmp(__SYCL_PROPERTY_SET_DEVICELIB_BF16_TYPE,
                                     ImgPS->Name);
                    });
```
Not a strong preference, so feel free to ignore.

Reply:
Hi, @steffenlarsen 
I prefer to keeping original version of the code snippet. Currently, we only handle bf16 devicelib property in skipEmptyImage but we may add new checking logic in this function, the previous code seems to be easier for extending the logic.
Thanks very much.